cirq/ops/common_gates.py

```diff
@@ -116,6 +116,30 @@ def _trace_distance_bound_(self) -> Optional[float]:
             return None
         return abs(np.sin(self._exponent * 0.5 * np.pi))
 
+    def controlled(self,
+                   num_controls: int = None,
+                   control_values: Optional[Sequence[
+                       Union[int, Collection[int]]]] = None,
+                   control_qid_shape: Optional[Tuple[int, ...]] = None
+                  ) -> raw_types.Gate:
+        """
+        Specialize controlled method for XPowGate to construct a corresponding
+        controlled CNotPowGate, if applicable.
+        Specialization only happens when the last specified control (which acts
+        first semantically) is a default-type control qubit.
+        """
+        result = super().controlled(num_controls, control_values,
+                                    control_qid_shape)
+        if (isinstance(result, controlled_gate.ControlledGate) and
+                result.control_values[-1] == (1,) and
+                result.control_qid_shape[-1] == 2):
+            return cirq.CNotPowGate(exponent=self._exponent,
+                                    global_shift=self._global_shift).controlled(
+                                        result.num_controls() - 1,
```

Comment:
What happens if num_controls() == 1?  Will this still work correctly?
Is this handled by the super() method?

Reply:
Yeah, the purpose of calling the super method is to delegate all the special case handling to it.